src/domain/backbone/backbone.repository.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { faker } from '@faker-js/faker';
+import { HttpException } from '@nestjs/common';
+import { JsonSchemaService } from '../../common/schemas/json-schema.service';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import backboneFactory from '../balances/entities/__tests__/backbone.factory';
+import { ITransactionApi } from '../interfaces/transaction-api.interface';
+import { ITransactionApiManager } from '../interfaces/transaction-api.manager.interface';
+import { BackboneRepository } from './backbone.repository';
+import { Backbone } from './entities/backbone.entity';
+
+const BACKBONE: Backbone = backboneFactory();
+
+const transactionApi = {
+  getBackbone: jest.fn().mockResolvedValue(BACKBONE),
+} as unknown as ITransactionApi;
+
+const transactionApiManager = {
+  getTransactionApi: jest.fn().mockResolvedValue(transactionApi),
+} as unknown as ITransactionApiManager;
+
+const validationErrorFactory = {
+  from: jest.fn().mockReturnValue(new HttpException('testErr', 500)),
+} as unknown as ValidationErrorFactory;
```

Comment:
I know that we have some tests with similar setup but what is the advantage of mocking the return value here?

In my opinion it's usually more readable if the in test you see the setup, action, assertions. In this case part of the setup happens outside of the test.

Reply:
I agree, I usually set here the "default" behaviour to avoid repeating it in each test, but maybe it's better to have it defined per test, specially here where we're just mocking one function. 👍🏻

I changed it for this test class in [d376ff0](https://github.com/5afe/safe-client-gateway-nest/pull/82/commits/d376ff082174dc6624e1eeb6ec396f4bfbc36045), we can start using this convention when modifying test classes in other PRs 🙂

Reply:
> I agree, I usually set here the "default" behaviour to avoid repeating it in each test, but maybe it's better to have it defined per test, specially here where we're just mocking one function. 👍🏻

I do think that repetition in tests is ok – so that you can look at a single test and understand the setup and the assertions just from the test block.

If someone wants to write more test cases they can assume a "clean" setup for the test (ie.: no assumptions).